controllers/admin/AdminImportController.php

```diff
@@ -1312,109 +1773,125 @@ protected function categoryImportOne($info, $default_language_id, $id_lang, $for
             ), 'Admin.Advparameters.Notification');
         }
         $res = false;
-        if (($field_error = $category->validateFields(UNFRIENDLY_ERROR, true)) === true &&
-            ($lang_field_error = $category->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true && empty($this->errors)) {
+        if (($field_error = $category->validateFields(UNFRIENDLY_ERROR, true)) === true
+            && ($lang_field_error = $category->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true
+            && empty($this->errors)
+        ) {
             $category_already_created = Category::searchByNameAndParentCategoryId(
-                $id_lang,
-                $category->name[$id_lang],
+                $idLang,
+                $category->name[$idLang],
                 $category->id_parent
             );
 
             // If category already in base, get id category back
             if ($category_already_created['id_category']) {
-                $cat_moved[$category->id] = (int)$category_already_created['id_category'];
-                $category->id = (int)$category_already_created['id_category'];
+                $catMoved[$category->id] = (int)$category_already_created['id_category'];
+                $category->id            = (int)$category_already_created['id_category'];
                 if (Validate::isDate($category_already_created['date_add'])) {
                     $category->date_add = $category_already_created['date_add'];
                 }
             }
 
             if ($category->id && $category->id == $category->id_parent) {
-                $this->errors[] = sprintf(
-                    $this->trans('A category cannot be its own parent. The parent category ID is either missing or unknown (ID: %1$s).', array(), 'Admin.Advparameters.Notification'),
-                    (isset($info['id']) && !empty($info['id']))? $info['id'] : 'null'
+                $this->errors[] = $this->trans(
+                    'A category cannot be its own parent. The parent category ID is either missing or unknown '
+                        . '(ID: %1$s).',
+                    array(!empty($categoryId) ? $categoryId : 'null'),
+                    'Admin.Advparameters.Notification'
                 );
+
                 return;
             }
 
             /* No automatic nTree regeneration for import */
             $category->doNotRegenerateNTree = true;
 
             // If id category AND id category already in base, trying to update
-            $categories_home_root = array(Configuration::get('PS_ROOT_CATEGORY'), Configuration::get('PS_HOME_CATEGORY'));
-            if ($category->id &&
-                $category->categoryExists($category->id) &&
-                !in_array($category->id, $categories_home_root) &&
-                !$validateOnly) {
+            if ($category->id
+                && $category->categoryExists($category->id)
+                && !in_array($category->id, $baseCategories)
```

Comment:
Indexes are not used here. Just a plain in_array() to check current (soon to be created) category id against known existing categories ids.